rclcpp/src/rclcpp/node.cpp

```diff
@@ -146,6 +194,20 @@ Node::Node(
   sub_namespace_(""),
   effective_namespace_(create_effective_namespace(this->get_namespace(), sub_namespace_))
 {
+  // we have got what we wanted directly from the overrides,
+  // but declare the parameters anyway so they are visible.
+  rclcpp::detail::declare_qos_parameters(
+    rclcpp::QosOverridingOptions
+  {
+    QosPolicyKind::Depth,
+    QosPolicyKind::Durability,
+    QosPolicyKind::History,
+    QosPolicyKind::Reliability,
+  },
+    node_parameters_,
+    node_topics_->resolve_topic_name("/parameter_events"),
+    options.parameter_event_qos(),
+    rclcpp::detail::PublisherQosParametersTraits{});
```

Comment:
The indentation is weird here, is that uncrustify doing that?

I'd expect:

```suggestion
  rclcpp::detail::declare_qos_parameters(
    rclcpp::QosOverridingOptions{
      QosPolicyKind::Depth,
      QosPolicyKind::Durability,
      QosPolicyKind::History,
      QosPolicyKind::Reliability,
    },
    node_parameters_,
    node_topics_->resolve_topic_name("/parameter_events"),
    options.parameter_event_qos(),
    rclcpp::detail::PublisherQosParametersTraits{});
```

Reply:
I agree with your suggestion, but this is a problem in our uncrustify settings.
I also tried:

```cpp
  rclcpp::detail::declare_qos_parameters(
    rclcpp::QosOverridingOptions({
      QosPolicyKind::Depth,
      QosPolicyKind::Durability,
      QosPolicyKind::History,
      QosPolicyKind::Reliability,
    }),
    ...
```

without success.

I preferred to follow uncrustify requested style, but I can also add a `INDENT-OFF` comment.

Reply:
Nah it's fine as-is. Oh well 🤷‍♂️